crates/router/src/core/errors/customers_error_response.rs

```diff
@@ -0,0 +1,50 @@
+use http::StatusCode;
+
+#[derive(Debug, Clone, serde::Serialize)]
+pub enum CustomersErrorType {
+    ObjectNotFound,
+    InvalidRequestError,
+    InternalServerError,
+}
+
+#[derive(Debug, Clone, router_derive::ApiError)]
+#[error(error_type_enum = CustomersErrorType)]
```

Comment:
Yes, This is a custom macro mandatory for `error` macro attribute. I guess this will derive thiserror

Reply:
Also we already use `CE` for connector errors,

the error codes are horizontally layered rather than domain...
instead of the error codes specifying connector_error, payments_error, refunds_error.

instead we specify invalid_request (IR), hyperswitch_error (HE), connector_error(CE) etc...

can we use `thiserror::error` macro here instead?